src/main/java/mate.academy/lib/Injector.java

```diff
@@ -1,13 +1,69 @@
 package mate.academy.lib;
 
+import java.lang.reflect.Constructor;
+import java.lang.reflect.Field;
+import java.util.HashMap;
+import java.util.Map;
+import mate.academy.service.impl.ImplementationFinder;
+
 public class Injector {
     private static final Injector injector = new Injector();
+    private final Map<Class<?>, Object> instances = new HashMap<>();
 
     public static Injector getInjector() {
         return injector;
     }
 
     public Object getInstance(Class<?> interfaceClazz) {
-        return null;
+        if (interfaceClazz.isAnnotationPresent(Component.class)) {
+            Object clazzImplementationInstance = null;
+            Class<?> clazz = findImplementation(interfaceClazz);
+            Field[] declaredFields = clazz.getDeclaredFields();
+            for (Field field : declaredFields) {
+                if (field.isAnnotationPresent(Inject.class)) {
+                    Object fieldInstance = getInstance(field.getType());
+                    clazzImplementationInstance = createNewInstance(clazz);
```

Comment:
Creating a new instance of the class should be done before entering the loop that sets the fields. Otherwise, you will overwrite the `clazzImplementationInstance` with a new instance on each iteration where an `@Inject` field is found, leading to loss of already set fields and unnecessary object creation.